internal/clusterfeature/clusterfeatureadapter/cadence_feature_operation_dispatcher.go

```diff
@@ -50,8 +50,8 @@ func (d CadenceFeatureOperationDispatcher) DispatchApply(ctx context.Context, cl
 }
 
 // DispatchDeactivate dispatches a Deactivate request to a feature manager asynchronously
-func (d CadenceFeatureOperationDispatcher) DispatchDeactivate(ctx context.Context, clusterID uint, featureName string) error {
-	return d.dispatchOperation(ctx, workflow.OperationDeactivate, clusterID, featureName, nil)
+func (d CadenceFeatureOperationDispatcher) DispatchDeactivate(ctx context.Context, clusterID uint, featureName string, spec clusterfeature.FeatureSpec) error {
```

Comment:
What??? Why? Not a fan of extending the interface unless *absolutely* necessary.

Reply:
Unfortunately, we need input fields (specially `secretID` to communicate with Vault) to deactivate Vault feature

Reply:
During activation the spec is new, so we need to pass it to the dispatcher. In case of deactivation, the dispatcher could fetch it from the database.

Reply:
For async operations it's better to have a snapshot of the data the operation should be working with. Trying to load from the database inside the operation may result in inconsistencies.